### PR TITLE
[FIX] for GPX and M8B export thread queue and cache paths

### PR DESCRIPTION
**Pull request by rksh**
- at-most-one Thread queue was failing to respect job classes
- cache paths were missing a "/"
- added cancel-ability (not yet wired)

### CODE REVIEW

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/GpxManagementActivity.java [hunk 1]
```diff
@@ -177,8 +177,8 @@ public void handleDialog(int dialogId) {
             case EXPORT_GPX_DIALOG: {
                 if (!exportRouteGpxFile(exportRouteId)) {
                     Logging.warn("Failed to export gpx.");
-                    WiGLEToast.showOverFragment(this, R.string.error_general,
-                            getString(R.string.gpx_failed));
+                    //WiGLEToast.showOverFragment(this, R.string.error_general,
+                    //        getString(R.string.gpx_failed));
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/GpxManagementActivity.java:181]: Should the user still be notified of this?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/UniqueTaskExecutorService.java [hunk 2]
```diff
@@ -21,25 +25,89 @@ public UniqueTaskExecutorService(int nThreads) {
...
 
     @Override
     protected void beforeExecute(Thread t, Runnable r) {
-        Logging.debug("===>EXECUTING: "+r.getClass());
-        current = r;
+        if (r instanceof WrappedTask) {
+            WrappedTask wt = (WrappedTask)r;
+            Logging.debug("===>EXECUTING: " + wt.getWrappedClass());
+            current = r;
+        }
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/UniqueTaskExecutorService.java:42]: Should there be a log if this elses for some reason?

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/UniqueTaskExecutorService.java [hunk 3]
```diff
@@ -21,25 +25,89 @@ public UniqueTaskExecutorService(int nThreads) {
...
     protected void afterExecute(Runnable r, Throwable t) {
-        jobClasses.remove(r.getClass());
-        current = null;
-        Logging.debug("<===EXECUTED: "+r.getClass());
+        if (r instanceof  WrappedTask) {
+            WrappedTask wt = (WrappedTask)r;
+            jobClasses.remove(wt.getWrappedClass());
+            current = null;
+            Logging.debug("<===EXECUTED: " + wt.getWrappedClass());
+        }
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/background/UniqueTaskExecutorService.java:52]: Same else notion.

**Review by bobzilladev — APPROVED** [review]
Seems ok, uhtu has been the one digging in on this goo though.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/GpxManagementActivity.java (hunk 1 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/GpxManagementActivity.java:181]: the only case where we return false turns out to be the one we can (and are) more specific about inside the method (and the specific followed by the non-specific is annoying)

**Review by uhtu — APPROVED** [review]
lgtm (modulo bobz' else clause notes)

**Pull request merged**